pywren/executor.py

```diff
@@ -176,27 +166,25 @@ def map(self, func, iterdata, extra_env = None, extra_meta = None,
         host_job_meta = {}
 
         pool = ThreadPool(invoke_pool_threads)
-        callset_id = s3util.create_callset_id()
+        callset_id = wrenutil.create_callset_id()
+        data = list(iterdata)
```

Comment:
The `data=list(iterdata)` line does appear twice here. @ooq ?

![screen shot 2017-06-30 at 11 00 16 pm](https://user-images.githubusercontent.com/7637700/27759550-2719c3d6-5de8-11e7-95f0-d486705b6f39.png)